qutebrowser/config/configutils.py

```diff
@@ -139,17 +168,28 @@ def remove(self, pattern: urlmatch.UrlPattern = None) -> bool:
         If no matching pattern was found, False is returned.
         """
         self._check_pattern_support(pattern)
-        old_len = len(self._values)
-        self._values = [v for v in self._values if v.pattern != pattern]
-        return old_len != len(self._values)
+        if pattern not in self._vmap:
+            return False
+        host = Values._pattern_to_host(pattern)
+        scoped_value = self._vmap[pattern]
+        # If we error here, that means domain_map and vmap are out of sync,
+        # report a bug!
+        assert host in self._domain_map
+        self._domain_map[host].remove(scoped_value)
+        del self._vmap[pattern]
+        return True
 
     def clear(self) -> None:
         """Clear all customization for this value."""
-        self._values = []
+        self._vmap.clear()
+        self._domain_map.clear()
+        self._scoped_id = 0
 
     def _get_fallback(self, fallback: typing.Any) -> typing.Any:
```

Comment:
(Although this is from old code) should the type of `fallback` be `bool` instead of `Any`?

Reply:
Indeed, thanks! Fixed in de3e09673bcaed7a1db2e400543adba2b3186db0.